### PR TITLE
Add support for fedora-coreos

### PR DESCRIPTION
**Pull request by martinpitt**
https://github.com/cockpit-project/bots/pull/218 has landed now.

 * [x] work around NetworkManager reload failure: https://bugzilla.redhat.com/show_bug.cgi?id=1771422 : https://github.com/cockpit-project/bots/pull/232
 * [x] work around pwscore failure: https://github.com/coreos/fedora-coreos-config/issues/230 : https://github.com/cockpit-project/bots/pull/232
 * [x] Fix check-journal binary log tests to not require Python: https://github.com/cockpit-project/cockpit/pull/13135
 * [x] fix check-pages: https://github.com/cockpit-project/cockpit/pull/13136

### CODE REVIEW

**Comment by martinpitt** [comment]
Only [104 failures](https://209.132.184.41:8493/logs/pull-13110-20191107-161111-946a5b0b-cockpit-project-cockpit--fedora-coreos/log.html), easy peasy! :-)

**Comment by martinpitt** [comment]
Next round -- https://github.com/cockpit-project/bots/pull/232 landed with workarounds for two bugs in FCOS itself, and I pushed some more test adaptions. This should clear the vast majority of failures, let's see what's left now.

**Comment by martinpitt** [comment]
ok, 104 → [23 failures](https://209.132.184.41:8493/logs/pull-13110-20191112-135454-131437ca-cockpit-project-cockpit--fedora-coreos/log.html), that's something already.

**Comment by martinpitt** [comment]
[CPU security mitigations](https://209.132.184.41:8493/logs/pull-13110-20191112-155151-8550cb42-cockpit-project-cockpit--fedora-coreos/log.html#195-2) is a rather hard done, as there's no grubby on Fedora CoreOS. I need to find a way to update the kernel parameters there.

**Comment by martinpitt** [comment]
I filed #13138 about the outstanding CPU mitigations. This affects all OSes, not just CoreOS, but this spotted the problem. In the meantime I see no reason to block this PR on that, as that's independent.

File: test/verify/check-active-pages [hunk 1]
```diff
@@ -32,7 +32,7 @@ class TestActivePages(MachineCase):
         # Preloading via manifests only happens with #12534
         if m.image in [ "rhel-8-1-distropkg" ]:
             n_extra_preloaded = 0
-        elif "atomic" in m.image:
+        elif m.atomic_image or "coreos" in m.image:
```
Comment (marusak) [inline test/verify/check-active-pages]: This could be `elif m.ostree_image`

File: test/verify/check-keys [hunk 2]
```diff
@@ -166,7 +166,7 @@ class TestKeys(MachineCase):
             b.click(selector + " .onoff-ct input")
 
         # Operating systems where auto loading doesn't work
-        auto_load = "atomic" not in m.image
+        auto_load = not m.atomic_image and "coreos" not in m.image
```
Comment (marusak) [inline test/verify/check-keys]: `auto_load = not m.ostree_image`

File: test/verify/check-system-info [hunk 3]
```diff
@@ -501,7 +504,7 @@ class TestSystemInfo(MachineCase):
 
         # Memory details should be shown for generic QEMU memory
         # demidecode not available on certain images
-        if m.image not in ['fedora-atomic']:
```
Comment (marusak) [inline test/verify/check-system-info]: this was not skipped on `rhel-atomic` before, but is now.

Comment (marusak) [reply test/verify/check-system-info]: Ah, but that is `rhel-7.7` based so we test it only in stable rhel-7 branches. Nevermind :)

File: test/verify/check-active-pages (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-active-pages]: Ah, that was clearly a rebase error, I'll clean it up.

File: test/verify/check-system-info (hunk 3 above)
Comment (martinpitt) [reply test/verify/check-system-info]: Right, I found it easier that way. Also, right now on master we don't test *any* atomic image any more. After this PR lands, I'll clean up the *-atomic special cases.

**Comment by martinpitt** [comment]
@marusak: I fixed the little rebase errors, thanks for spotting! (No effective changes, but cleaner history). `testCPUSecurityMitigations` broke for two different reasons, hence there are two commits. (1) FCOS enables "nosmt" by default, so the test needs to get taught to deal with that. Then it already gets much further. And (2) we don't parse `mitigations=nosmt`, which is issue #13138. Fixing (2) is rather involved, and I don't want to bundle it to this PR. Thus the second commit still skips the test.

**Pull request merged**